tests/FilterTest.php

```diff
@@ -1,40 +1,26 @@
 <?php
 
+declare(strict_types = 1);
+
 namespace Lambdish\Phunctional\Tests;
 
-use PHPUnit_Framework_TestCase;
+use PHPUnit\Framework\TestCase;
 use function Lambdish\Phunctional\filter;
 
-class FilterTest extends PHPUnit_Framework_TestCase
+final class FilterTest extends TestCase
 {
     /** @test */
-    public function it_should_filter_a_collection_keeping_the_indexes()
+    public function it_should_filter_a_collection_keeping_the_indexes(): void
```

Comment:
Can we test the asserting with an array and also with an iterator?

$actual = new ArrayIterator(['one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5]);
$pred   = $this->evenByValueFilterer();

$this->assertSame(['two' => 2, 'four' => 4], filter($pred, $actual));